Tasks/CopyFiles/task.json

```diff
@@ -0,0 +1,87 @@
+﻿{
+  "id": "5BFB729A-A7C8-4A78-A7C3-8D717BB7C13C",
+  "name": "CopyFiles",
+  "friendlyName": "Copy Files",
+  "description": "Copy files from source folder to target folder using minimatch patterns",
+  "helpMarkDown": "[More Information](http://go.microsoft.com/fwlink/?LinkID=)",
+  "category": "Utility",
+  "visibility": [
+    "Build"
+  ],
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 1,
+    "Minor": 0,
+    "Patch": 0
+  },
+  "demands": [
+  ],
+  "minimumAgentVersion": "1.91.0",
+  "groups": [
+        {
+            "name": "advanced",
+            "displayName": "Advanced",
+            "isExpanded": false
+        }
+    ],
+  "inputs": [
+    {
+      "name": "SourceFolder",
+      "type": "string",
+      "label": "Source Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Source folder to apply copy patterns to"
+    },  
+    {
+      "name": "Contents",
+      "type": "multiLine",
+      "label": "Contents",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "File paths to include as part of the copy. Supports multiple lines or minimatch patterns. [More Information](http://go.microsoft.com/fwlink/?LinkID=)"
+    },
+    {
+      "name": "TargetFolder",
+      "type": "string",
+      "label": "Target Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Target folder files will copy to"
+    },
+    {
+      "name": "CleanTargetFolder",
+      "type": "boolean",
+      "label": "Clean Target Folder",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Delete all existing files in target folder before copy.",
+      "groupName":"advanced"
+    },
+    {
+      "name": "OverWrite",
+      "type": "boolean",
+      "label": "Over Write",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Relace existing file in target folder",
+      "groupName":"advanced"
+    },
+    {
+      "name": "UseCommonRoot",
+      "type": "boolean",
+      "label": "Use Common Root",
+      "defaultValue": "false",
```

Comment:
You should write up a document on how to use this task as I think it could be confusing for people.

From: TingluoHuang [mailto:notifications@github.com]
Sent: Tuesday, November 3, 2015 8:22 AM
To: Microsoft/vso-agent-tasks <vso-agent-tasks@noreply.github.com>
Cc: Chris Patterson <Chris.Patterson@microsoft.com>
Subject: Re: [vso-agent-tasks] Users/tihuang/newartifact (#698)


In Tasks/CopyFiles/task.json<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMicrosoft%2fvso-agent-tasks%2fpull%2f698%23discussion_r43769194&data=01%7c01%7cChris.Patterson%40microsoft.com%7c5a8c11f187174d70c15908d2e46af686%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=KDL14STH6%2fhoOxBvJ8Zv5F%2bgScOarodIQbXNb9F%2bF4Y%3d>:

> +      "groupName":"advanced"

> +    },

> +    {

> +      "name": "OverWrite",

> +      "type": "boolean",

> +      "label": "Over Write",

> +      "defaultValue": "false",

> +      "required": false,

> +      "helpMarkDown": "Relace existing file in target folder",

> +      "groupName":"advanced"

> +    },

> +    {

> +      "name": "UseCommonRoot",

> +      "type": "boolean",

> +      "label": "Use Common Root",

> +      "defaultValue": "false",

Short the path on target folder by remove the common root of files need copied.
Ex.
source: c:\a\b\c\test1.txt
c:\a\b\c\test2.txt
c:\a\b\c\test3.txt
SourceFolder = c:\a
TargetFolder = d:\t
With using common root.
copy result:
d:\t\test1.txt
d:\t\test2.txt
d:\t\test3.txt
Otherwise:
d:\t\b\c\test1.txt
d:\t\b\c\test2.txt
d:\t\b\c\test3.txt

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMicrosoft%2fvso-agent-tasks%2fpull%2f698%2ffiles%23r43769194&data=01%7c01%7cChris.Patterson%40microsoft.com%7c5a8c11f187174d70c15908d2e46af686%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=y%2fPMAQAa52LmdoUAqGu%2bl8kO6cZj3uygijNDmrhKrb8%3d>.